### ISSUE TITLE
1.19 blocks not being rendered

### ISSUE THREAD

**Issue by BenKennish** [overviewer/Minecraft-Overviewer#2046]
Just finished rendering a 1.19 map using Minecraft Overviewer 0.17.63 (97ef871) but the blocks new to 1.19 do not seem to be being rendered.  I downloaded the textures from https://overviewer.org/textures/1.19 into ~/.minecraft/versions/1.19.jar before starting the render.

The render process did not throw up any warnings or errors except for some:
"W Unable to get player name for UUID"  which are caused by our server using the Bedrock->Java proxy called Geyser - another issue I need to sort out sometime.

Here's a screenshot of the latest render...

![Screenshot 2022-06-16 104850](https://user-images.githubusercontent.com/1702146/174043865-8b7af0a3-4d34-4e1a-9f29-2794245a4899.png)

The 3 torches in a line just below the "white arrow on orange" marker thing are supposed to be on top of mud brick blocks and there is also a mud brick slab and mud brick steps close by that should be visible.   I'm not sure what the block that looks like a block enclosed by wooden fence panels is supposed to be but I think it's the skulk catalyst that I had there.  Also the black spaces I think should be skulk blocks.

Thanks in advance for any help :)

**Comment by CounterPillow** [comment]
I haven't added any 1.19 blocks yet, so this is expected

**Comment by BenKennish** [comment]
Ah I see.  Thanks for the info.  I checked the 1.19.jar textures file and it seemed larger than 1.18.2.jar so I assumed it had the new blocks inside.   Is there a way I could receive a notification when the 1.19 blocks have been added?  I may wait until that happens before adding the render to a cron job.  Cheers.

**Comment by CounterPillow** [comment]
If you have e-mail notifications on github enabled, you will get one when I close this issue or someone comments on it.

**Comment by IncredibleHolg** [comment]
To get the list of missing blocks, I added this to world.py
```python
    def _get_block(self, palette_entry):
        if ( palette_entry['Name'] not in self._blockmap):
            print (palette_entry['Name'] + "\n")
```

The filtered output, when I render my testworld, delivered this missing blocks:
( only the blocks placed are showing up )

* minecraft:mangrove_button
* minecraft:mangrove_door
* minecraft:mangrove_fence
* minecraft:mangrove_fence_gate
* minecraft:mangrove_leaves
* minecraft:mangrove_log
* minecraft:mangrove_planks
* minecraft:mangrove_pressure_plate
* minecraft:mangrove_propagule
* minecraft:mangrove_roots
* minecraft:mangrove_sign
* minecraft:mangrove_slab
* minecraft:mangrove_stairs
* minecraft:mangrove_trapdoor
* minecraft:mangrove_wood
* minecraft:stripped_mangrove_log
* minecraft:mud
* minecraft:mud_brick_slab
* minecraft:mud_brick_stairs
* minecraft:mud_brick_wall
* minecraft:mud_bricks
* minecraft:muddy_mangrove_roots
* minecraft:packed_mud

Btw, this old blocks also showed up
* minecraft:target
* minecraft:flowering_azalea
* minecraft:potted_cornflower
* minecraft:nether_sprouts

Is this list complete or are there more new blocks?

**Comment by IncredibleHolg** [comment]
Missing frog related blocks 
* minecraft:frogspawn
* minecraft:ochre_froglight
* minecraft:pearlescent_froglight
* minecraft:verdant_froglight

**Comment by monkeyworknet** [comment]
Commenting for email notification - thx everyone who is working on this.

**Comment by BenKennish** [comment]
Is anyone working on this?  If not, I'm happy to give it a go.  I'm not a Python dev (primarily PHP) but i've done a bit and it seems that this might not be a hugely taxing task?

**Comment by granis** [comment]
> Is this list complete or are there more new blocks?

Current overviewer covers 87% of blocks from 1.19 (117 blocks from a total of 933).

- minecraft:mangrove_planks
- minecraft:mangrove_propagule
- minecraft:mangrove_log
- minecraft:mangrove_roots
- minecraft:muddy_mangrove_roots
- minecraft:stripped_mangrove_log
- minecraft:mangrove_wood
- minecraft:stripped_mangrove_wood
- minecraft:mangrove_leaves
- minecraft:moving_piston
- minecraft:mangrove_sign
- minecraft:mangrove_wall_sign
- minecraft:mangrove_pressure_plate
- minecraft:mangrove_trapdoor
- minecraft:packed_mud
- minecraft:mud_bricks
- minecraft:mud_brick_stairs
- minecraft:potted_mangrove_propagule
- minecraft:potted_dandelion
- minecraft:potted_cornflower
- minecraft:potted_lily_of_the_valley
- minecraft:potted_wither_rose
- minecraft:mangrove_button
- minecraft:skeleton_skull
- minecraft:wither_skeleton_skull
- minecraft:zombie_head
- minecraft:player_head
- minecraft:creeper_head
- minecraft:dragon_head
- minecraft:mangrove_stairs
- minecraft:barrier
- minecraft:light
- minecraft:mangrove_slab
- minecraft:mud_brick_slab
- minecraft:mangrove_fence_gate
- minecraft:mangrove_fence
- minecraft:mangrove_door
- minecraft:end_gateway
- minecraft:structure_void
- minecraft:turtle_egg
- minecraft:dead_tube_coral
- minecraft:dead_brain_coral
- minecraft:dead_bubble_coral
- minecraft:dead_fire_coral
- minecraft:dead_horn_coral
- minecraft:dead_tube_coral_fan
- minecraft:dead_brain_coral_fan
- minecraft:dead_bubble_coral_fan
- minecraft:dead_fire_coral_fan
- minecraft:dead_horn_coral_fan
- minecraft:dead_tube_coral_wall_fan
- minecraft:dead_brain_coral_wall_fan
- minecraft:dead_bubble_coral_wall_fan
- minecraft:dead_fire_coral_wall_fan
- minecraft:dead_horn_coral_wall_fan
- minecraft:conduit
- minecraft:mud_brick_wall
- minecraft:nether_sprouts
- minecraft:target
- minecraft:potted_crimson_fungus
- minecraft:potted_warped_fungus
- minecraft:potted_crimson_roots
- minecraft:potted_warped_roots
- minecraft:candle
- minecraft:white_candle
- minecraft:orange_candle
- minecraft:magenta_candle
- minecraft:light_blue_candle
- minecraft:yellow_candle
- minecraft:lime_candle
- minecraft:pink_candle
- minecraft:gray_candle
- minecraft:light_gray_candle
- minecraft:cyan_candle
- minecraft:purple_candle
- minecraft:blue_candle
- minecraft:brown_candle
- minecraft:green_candle
- minecraft:red_candle
- minecraft:black_candle
- minecraft:candle_cake
- minecraft:white_candle_cake
- minecraft:orange_candle_cake
- minecraft:magenta_candle_cake
- minecraft:light_blue_candle_cake
- minecraft:yellow_candle_cake
- minecraft:lime_candle_cake
- minecraft:pink_candle_cake
- minecraft:gray_candle_cake
- minecraft:light_gray_candle_cake
- minecraft:cyan_candle_cake
- minecraft:purple_candle_cake
- minecraft:blue_candle_cake
- minecraft:brown_candle_cake
- minecraft:green_candle_cake
- minecraft:red_candle_cake
- minecraft:black_candle_cake
- minecraft:amethyst_cluster
- minecraft:tinted_glass
- minecraft:sculk_sensor
- minecraft:sculk
- minecraft:sculk_vein
- minecraft:sculk_catalyst
- minecraft:sculk_shrieker
- minecraft:azalea
- minecraft:flowering_azalea
- minecraft:big_dripleaf
- minecraft:big_dripleaf_stem
- minecraft:small_dripleaf
- minecraft:mud
- minecraft:potted_azalea_bush
- minecraft:potted_flowering_azalea_bush
- minecraft:ochre_froglight
- minecraft:verdant_froglight
- minecraft:pearlescent_froglight
- minecraft:frogspawn
- minecraft:reinforced_deepslate

**Comment by BenKennish** [comment]
Is it that the textures are in the .jar file but the code needs updating to handle each new block?  Or do the textures need adding to the .jar file so the code can access them?  Or both? 😆

**Comment by BenKennish** [comment]
@CounterPillow ?

**Comment by LieuweWestra** [comment]
@BenKennish Every block needs to be added to the code individually in multiple places. For simple blocks this is relatively easy, for blocks with complicated models this is a lot of work. But the code is pretty straight forward and Pillow is pretty good at reviewing, so you could give it a try.

**Comment by CounterPillow** [comment]
I added mud, packed mud, mud bricks, mud brick slabs, mangrove leaves, mangrove roots, mangrove logs and stripped mangrove logs.

I still hate spending time on this because it's something that wouldn't need time invested if someone were to rewrite Overviewer to actually do block models, but I guess bandaid fixes is all I have the motivation for.

**Comment by LieuweWestra** [comment]
Created PR for some more 119 wood blocks. https://github.com/overviewer/Minecraft-Overviewer/pull/2048
Coulnt figure out what I was doing wrong with the slabs so thats for another time.

**Comment by Teenyplx** [comment]
Any progress on this? Looking at the above from lieuwestra it seems like he had a more in-depth fix, but it seems to still be pending a merge? (Not very proficient at Git so i'm unsure)

**Comment by IncredibleHolg** [comment]
@lieuwestra 
You are using match/case in your code. This requires python >= 3.10., which will break the support for Ubuntu LTE 20.4 
(EOL 2025), since this only support python 3.8.
Upgrading to Ubuntu 22.4 will not be an option for every installation.
Can you replace the match/case in `overviewer_core/textures.py` with an python 3.8 compatible construct?

**Comment by lieuwestra** [comment]
@IncredibleHolg fixed

**Comment by IncredibleHolg** [comment]
Hi,
my main server world is very old and big ( started in 1.8 ), consequently there may be chunks never visited in the new versions.
When I try to render this with pull/2054/head, this fails with
```console
E Could not render chunk 62,22 for some reason. This is likely a render primitive option error
E Full error was:
Traceback (most recent call last):
  File "./overviewer_core/tileset.py", line 1105, in _render_rendertile
    c_overviewer.render_loop(
  File "./overviewer_core/world.py", line 1930, in get_chunk
    retval = super(CachedRegionSet, self).get_chunk(x,z)
  File "./overviewer_core/world.py", line 1743, in get_chunk
    return self._r.get_chunk(x,z)
  File "./overviewer_core/world.py", line 1604, in get_chunk
    (blocks, data) = self._get_blockdata_v113(section, longarray_unpacker)
  File "./overviewer_core/world.py", line 1399, in _get_blockdata_v113
    translated_blocks[i], translated_data[i] = self._get_block(key)
  File "./overviewer_core/world.py", line 1079, in _get_block
    axis = palette_entry['Properties']['axis']
KeyError: 'axis'
```

looks like chunks in 1.13 format.
Btw: this renders fine in the current master branch.

I will try to narrow down the source.

**Comment by lieuwestra** [comment]
@IncredibleHolg There might be hay bales without the axis property. I've fixed it by using a default values if the axis is missing. Should work now.

**Comment by IncredibleHolg** [comment]
Hi @lieuwestra,
thanks for your change.

The actual render finishes now without any error.
But when I call --genpoi, this error pops up:

```console
This is the error that occurred:
multiprocessing.pool.RemoteTraceback:
"""
Traceback (most recent call last):
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 48, in mapstar
    return list(map(*args))
  File "./Minecraft-Overviewer/overviewer_core/aux_files/genPOI.py", line 132, in parseBucketChunks
    data = rset.get_chunk(b[0], b[1])
  File "./Minecraft-Overviewer/overviewer_core/world.py", line 1605, in get_chunk
    (blocks, data) = self._get_blockdata_v118(section, longarray_unpacker)
  File "./Minecraft-Overviewer/overviewer_core/world.py", line 1375, in _get_blockdata_v118
    translated_blocks[i], translated_data[i] = self._get_block(key)
  File "./Minecraft-Overviewer/overviewer_core/world.py", line 1020, in _get_block
    (block, data) = modelblocks[slab_to_double[key]]
KeyError: 'minecraft:smooth_stone'
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "./Minecraft-Overviewer/overviewer.py", line 653, in <module>
    ret = main()
  File "./Minecraft-Overviewer/overviewer.py", line 168, in main
    g.genPOI.main()
  File "./Minecraft-Overviewer/overviewer_core/aux_files/genPOI.py", line 589, in main
    handleEntities(rset, config, args.config, rset_filters, markers)
  File "./Minecraft-Overviewer/overviewer_core/aux_files/genPOI.py", line 218, in handleEntities
    results = pool.map(parseBucketChunks, ((buck, rset, filters) for buck in buckets))
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 364, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 771, in get
    raise self._value
KeyError: 'minecraft:smooth_stone'
```
Looks like this is an unconverted 1.18 region.

The POI function searches for banners, no clue why smooth_stone is an issue

```python
global json
import json
def bannerFilter(poi):
    if poi['id'] == 'minecraft:banner':
        if 'CustomName' in poi.keys():
            cnstring = poi['CustomName']
            print (cnstring)
            cname = json.loads(cnstring)
            if 'text' in cname.keys():
                print(cname['text'])
                print(poi)
                return (cname['text'])

renders["render1"] = {
        'world': 'Nitrado vanilla',
        'title': 'Nitrado Vanilla',
        'markers': [dict(name="Banner", filterFunction=bannerFilter, icon="markers/marker_town.png" )]
}
```

**Comment by lieuwestra** [comment]
@IncredibleHolg good to know the regular redner is fine now. Are all expected blocks rendering properly?
The problem you described was caused by how double slabs get handled. I've fixed it.

**Comment by BenKennish** [comment]
Are these fixes are being applied to the master branch?   Could I consider this branch as "generally" stable and do a git pull before each daily render of my world or do you think that would be unwise?

**Comment by IncredibleHolg** [comment]
This is still in the state of a pull request and not is the master branch.

If you want to test it, you can follow [this instruction on devopscube.com](https://devopscube.com/checkout-git-pull-request/)

More testing will be helpful to get this merged into master.

**Comment by Teenyplx** [comment]
Just to be clear, this is the system that will read blocks dynamically and insert them into the code automatically right? So there is no longer a manual need to insert blocks in the future?

**Comment by IncredibleHolg** [comment]
Depends on the block.
If you check e.g. the lantern or bamboo, this is stiched togethe cutting out differnt parts of the texture.
Since The overwiewer is using very tiny images, this is hard to automate without ending up with an unrecognizable mess.
( There is no "Level of Detail" texture in Minecraft )

But full block textures will be automated.

**Comment by lieuwestra** [comment]
@BenKennish You can use it and it won't irretrievable corrupt your render and I will keep it up to date with the main branch, but will probably go through a less stable phase if I can ever get some commitment from @CounterPillow to do a proper review. He can still decide dynamic rendering will never go to main.

Also I won't update the C rebuild trigger every commit so if you don't rebuild the c code in your update script this might still break between commits.

**Comment by IncredibleHolg** [comment]
@lieuwestra 
Finished a complete map update this night.Worked fine, POI generated as expected.
New blocks example: usage of mangrove planks
![grafik](https://user-images.githubusercontent.com/41533138/182536816-ab694309-82e5-4a50-821d-33148bda402c.png)

**Comment by BenKennish** [comment]
> This is still in the state of a pull request and not is the master branch.
> 
> If you want to test it, you can follow [this instruction on devopscube.com](https://devopscube.com/checkout-git-pull-request/)
> 
> More testing will be helpful to get this merged into master.

I tried to follow the guide but didn't know what the name of the branch was that I needed to use so I assumed master but it failed..

```
$ git fetch origin pull/2054/head:master
fatal: Refusing to fetch into current branch refs/heads/master of non-bare repository
```

**Comment by lieuwestra** [comment]
I believe you have to add my repository to your remotes first, and then checkout the branch;
`git remote add https://github.com/lieuwestra/Minecraft-Overviewer.git`
`git fetch`
`git checkout read_models`

**Comment by IncredibleHolg** [comment]
@BenKennish 
Your syntax of the git fetch command is wrong.
You need to state the name of a non-existing local branch, which will be created and you can check this out afterwards.
```bash
git fetch original pull/2054/head:genericblockmodels
git checkout genericblockmodels
```

**Comment by BenKennish** [comment]
Ah that's great.  You typed `original` rather than `origin` but otherwise that worked fine.  Thank you.

**Comment by BenKennish** [comment]
However I cannot now pull with `genericblockmodels` branch checked out.   I get...

```
There is no tracking information for the current branch.
Please specify which branch you want to merge with.
See git-pull(1) for details.

    git pull <remote> <branch>

If you wish to set tracking information for this branch you can do so with:

    git branch --set-upstream-to=origin/<branch> genericblockmodels
```

I tried `git branch --set-upstream-to=origin/pull/2054/head genericblockmodels` and `git branch --set-upstream-to=origin/pull/2054 genericblockmodels` but neither worked.

**Comment by IncredibleHolg** [comment]
@BenKennish 
Given, the syntax, GIT uses, to access pull requests ist quite strange.
But here is a working seqeunce:
```bash
$ git clone https://github.com/overviewer/Minecraft-Overviewer
Cloning into 'Minecraft-Overviewer'...
$ cd Minecraft-Overviewer/
$ git fetch origin pull/2054/head:genericblockmodels
remote: Enumerating objects: 167, done.
remote: Counting objects: 100% (167/167), done.
remote: Compressing objects: 100% (58/58), done.
remote: Total 167 (delta 118), reused 155 (delta 109), pack-reused 0
Receiving objects: 100% (167/167), 170.92 KiB | 2.16 MiB/s, done.
Resolving deltas: 100% (118/118), completed with 4 local objects.
From https://github.com/overviewer/Minecraft-Overviewer
 * [new ref]         refs/pull/2054/head -> genericblockmodels
$ git checkout genericblockmodels
Switched to branch 'genericblockmodels'
```
As @lieuwestra mentioned, you may add his working clone, if you want to follow his changes.
```bash
$ git remote add lieuwestra https://github.com/lieuwestra/Minecraft-Overviewer.git
$ git fetch --all
Fetching origin
Fetching lieuwestra
remote: Enumerating objects: 70, done.
remote: Counting objects: 100% (70/70), done.
remote: Compressing objects: 100% (20/20), done.
remote: Total 70 (delta 53), reused 60 (delta 50), pack-reused 0
Unpacking objects: 100% (70/70), 46.21 KiB | 666.00 KiB/s, done.
From https://github.com/lieuwestra/Minecraft-Overviewer
 * [new branch]      complex_blocks -> lieuwestra/complex_blocks
 * [new branch]      master         -> lieuwestra/master
 * [new branch]      read_models    -> lieuwestra/read_models
$ git checkout read_models
Branch 'read_models' set up to track remote branch 'read_models' from 'lieuwestra'.
Switched to a new branch 'read_models'
```

On an additional branch, you can use ```git pull``` but on a branch created from a pull request, you have to delete the branch and do it all over again to get updates.

**Comment by BenKennish** [comment]
Ah sorry @lieuwestra I missed your post somehow.  Yeah that's great.  I've done that now and will keep pulling and using the `read_models` branch from than master of this repo until this is all merged in.  Thank you

**Comment by capnbfg** [comment]
@lieuwestra 
Thank you for implementing this!  I built it and tested on my 1.19 world, doing before and after comparisons between your update and the baseline 0.19.6 for many different materials.  Wherever I saw a difference, I compared to in-game visuals to confirm which render was correct.  In general, it seems that your version renders all stone, slate, and plank blocks in the correct orientation, whereas 0.19.6 often had the textures backwards.  I also confirmed that mangrove planks, steps, and fence all look correct after your update.  Leaf blocks are also improved in your version, with correct leaf pattern and color on dark oak and acacia trees, whereas 0.19.6 incorrectly rendered each of those with normal oak leaves.  So this update looks to me like an improvement for many materials.

I did find two issues that I wanted to report here.  It seems that your update produces the incorrect block orientation for mushroom blocks and log blocks.  Here are before/after comparisons for each of those two issues.

Notice the oak logs on the barn, first (correct) in 0.19.6 and second (incorrect) after your update:
![barn_0 19 6](https://user-images.githubusercontent.com/111255790/184779674-739f5e1c-546a-4000-8227-731c81d370e8.png)
![barn_lieuwestra](https://user-images.githubusercontent.com/111255790/184779676-76d86d93-cafa-481c-9bfc-0c946714d917.png)

And in this pair of images, notice the change in the mushrooms, first is 0.19.6 and second is after your update:
![shrooms_0 19 6](https://user-images.githubusercontent.com/111255790/184779874-f772f87f-483e-4b89-aca8-1b8c33628fcb.png)
![shrooms_lieuwestra](https://user-images.githubusercontent.com/111255790/184779877-8027b345-4698-4794-988b-08f27fbee545.png)

Thanks again.  I hope this helps.  I'm happy to retest if/when another update is available.

**Comment by lieuwestra** [comment]
@capnbfg
I've looked into this and there seems to be a problem with northdirection upper-right and lower-left. I'll have to look into this, might be a few days before it's fixed.

Also, won't your barn burn down with lightning rods on wood?

**Comment by capnbfg** [comment]
Thank you @lieuwestra, much appreciated.  It didn't occur to me to mention it before, but you are correct, my render did use northdirection upper-right.

Good question about the lightning rods.  It hasn't caused any fire yet.  If the Minecraft wiki page is correct, the ignition area is a 5x5 horizontal space centered on the lightning rod.  Since these lightning rods are above all the wood it shouldn't be an issue.

**Comment by IncredibleHolg** [comment]
@lieuwestra 
Since the 1.19.3 update your block model patch stoped working.
In textures,py find_texture_from_model:
textureset is {'texture': '#texture'}
face is "#texture"
This results in an infinitr recursion.
@BenKennish: is this your problem as well?

**Comment by BenKennish** [comment]
@IncredibleHolg very possibly.  I thought I'd tested it after switching the to master/main branch but I've just done that and run a render and got..

```
AttributeError: module 'overviewer_core.textures' has no attribute 'block_models'
```

**Comment by IncredibleHolg** [comment]
@BenKennish 
Did you recompile after switching the branch?
```bash
BRANCH=master
git checkout  $BRANCH && python3 setup.py build
```

**Comment by BenKennish** [comment]
Aha!  So I had forgotten to do this.  However, this gave me an error with an include(?) statement or something like that but then I remembered that I'm using Rocky Linux 8.7 (previously using CentOS 8) and I needed to compile with some flags to include some python-pillow stuff..

```
export tmpPillow=$(mktemp -d)
pushd $tmpPillow
yumdownloader --source python-pillow
rpm2cpio python-pillow-*.src.rpm | cpio -idmv
tar xf Pillow-*.tar.gz
popd
rm -rf build/
CPPFLAGS='-I'$(readlink -f ${tmpPillow}/Pillow-*/src/libImaging) python3 setup.py build
```
and this worked and the world render completed.  Of course, the render is now missing all the 1.19 blocks that were previously working when I was using @lieuwestra 's fork, but hopefully either that can be fixed or that work can be merged into the main project.

Thanks,

**Comment by BenKennish** [comment]
I've since found that I can avoid the python-pillow source extraction stuff by installing the `python3-pillow-devel` rpm package using `sudo dnf install python3-pillow-devel`.  Just an FYI in case that helps someone.7

**Comment by BenKennish** [comment]
Is the Overviewer project itself dead?  I opened this issue in June last year and the new blocks are still not supported.  The latest news post on the website (https://overviewer.org/) says "Overviewer 0.15 is out, supporting Minecraft 1.15 by CounterPillow on Dec. 27, 2019"

Don't get me wrong - I do appreciate the work that goes into projects like this but if the code is not really being maintained, I will try to look elsewhere for a similar solution.  Cheers.

**Comment by CounterPillow** [comment]
yes, Overviewer is no longer being maintained

**Comment by BenKennish** [comment]
ok thanks 👍   it's a shame because it's a great piece of software but i'm sure there are good reasons 🙂

**Comment by gnouts** [comment]
> yes, Overviewer is no longer being maintained

Would you mind share more about that ? I've spent the last few day setting up Overviewer, just to stumble on your comment.
Will you still accept PR ? Will the project be archived soon ? 
Shutting down a project is perfectly understandable, but would you mind writing that somewhere (in the frontpage readme) ?

**Comment by CounterPillow** [comment]
> Will you still accept PR ?

no

> Will the project be archived soon ?

no, the website and repository will remain accessible as-is.

> but would you mind writing that somewhere (in the frontpage readme) ?

Sure, I'll put a note in the README.

**Comment by ajh123** [comment]
The main README states ..
> PLEASE NOTE: Overviewer is currently unmaintained. PRs will not be merged and issues will not be addressed. The website and repository will remain online and accessible.

Instead please use the new working (“successor”) version of Overviewer supporting 1.20 worlds and textures.
https://github.com/GregoryAM-SP/The-Minecraft-Overviewer

**Comment by JoshuaMarti** [comment]
Yep, can confirm, most of the new blocks from 1.19 and 1.20 have been added and are working.
